internal/controllers/gateway/backendtlspolicy_controller.go

```diff
@@ -0,0 +1,297 @@
+// BackendTLSPolicy Controller - Indexers
+// -----------------------------------------------------------------------------
+
+const (
+	// backendTLSPolicyTargetRefIndexKey is the index key for BackendTLSPolicy objects by their target service reference.
+	// The value is the name of the service.
+	backendTLSPolicyTargetRefIndexKey = "backendtlspolicy-targetref"
+	// httpRouteParentRefIndexKey is the index key for HTTPRoute objects by their parent Gateway reference.
+	// The value is the namespace and name of the Gateway.
+	httpRouteParentRefIndexKey = "httproute-parentref"
+	// httpRouteBackendRefIndexKey is the index key for HTTPRoute objects by their backend service reference.
+	// The value is the namespace and name of the Service.
+	httpRouteBackendRefIndexKey = "httproute-backendref"
+)
+
+// indexBackendTLSPolicyOnTargetRef indexes BackendTLSPolicy objects by their target service reference.
+func indexBackendTLSPolicyOnTargetRef(obj client.Object) []string {
+	policy, ok := obj.(*gatewayapi.BackendTLSPolicy)
+	if !ok {
+		return []string{}
+	}
+
+	services := []string{}
+	for _, targetRef := range policy.Spec.TargetRefs {
+		if (targetRef.Group == "" || targetRef.Group == "core") && targetRef.Kind == "Service" {
+			services = append(services, string(targetRef.Name))
+		}
+	}
+	return services
+}
+
+// indexHTTPRouteOnParentRef indexes HTTPRoute objects by their parent Gateway references.
+func indexHTTPRouteOnParentRef(obj client.Object) []string {
+	httpRoute, ok := obj.(*gatewayapi.HTTPRoute)
+	if !ok {
+		return []string{}
+	}
+
+	gateways := []string{}
+	for _, parentRef := range httpRoute.Spec.ParentRefs {
+		// no need to check group and kind nilness, as they have a default value in case not specified
+		if *parentRef.Group == gatewayapi.V1Group && *parentRef.Kind == "Gateway" {
+			namespace := httpRoute.Namespace
+			if parentRef.Namespace != nil {
+				namespace = string(*parentRef.Namespace)
+			}
+			gateways = append(gateways, namespace+"/"+string(parentRef.Name))
+		}
+	}
+	return gateways
+}
+
+// indexHTTPRouteOnBackendRef indexes HTTPRoute objects by their backend service references.
+func indexHTTPRouteOnBackendRef(obj client.Object) []string {
+	httpRoute, ok := obj.(*gatewayapi.HTTPRoute)
+	if !ok {
+		return []string{}
+	}
+
+	services := []string{}
+	for _, rule := range httpRoute.Spec.Rules {
+		for _, backendRef := range rule.BackendRefs {
+			// no need to check group and kind nilness, as they have a default value in case not specified
+			if (*backendRef.Group != "core" && *backendRef.Group != "") || *backendRef.Kind != "Service" {
+				continue
+			}
+			namespace := httpRoute.Namespace
+			if backendRef.Namespace != nil {
+				namespace = string(*backendRef.Namespace)
+			}
+			services = append(services, namespace+"/"+string(backendRef.Name))
+		}
+	}
+	return services
+}
+
+// setupIndexers sets up the indexers for the BackendTLSPolicy controller.
+func setupIndexers(mgr ctrl.Manager) error {
```

Comment:
Please consider renaming this function as it's in the scope of a wider `gateway` package.
```suggestion
func setupBackendTLSPolicyIndices(mgr ctrl.Manager) error {
```